### DISCUSSION TITLE
Pagination with API

### DISCUSSION THREAD

**Discussion by castroben** [hub4j/github-api#1238] [Q&A]
I am using this module in Java to retrieve the number of contributors of a repository.The API returns 100 results for repositories that have a larger number of contributors than that. 

This makes me think that the API is not performing pagination. Is this true?

For reference I am using the GHRepositoryStatistics.ContributorStats class.

Finally, I have experienced very slow response times from this API. Is this normal?

**Comment by bitwiseman** [comment]
> The API returns 100 results for repositories that have a larger number of contributors than that.

No, pagination is used: https://github.com/hub4j/github-api/blob/dae92fceda572ef471c906d9d3104241c4c3bed6/src/main/java/org/kohsuke/github/GHRepositoryStatistics.java#L84-L91

However, the results must be generated (making the API slow) and it is possible the implementation is incorrect in some way.
